syntax = "proto3";

package kvstore;

service KVStore {
    rpc get (KeyRequest) returns (ValueResponse) {}
    rpc set (KeyValueRequest) returns (SetResponse) {}
    rpc getPrefix (KeyRequest) returns (stream ValueResponse) {}
}

message KeyRequest {
    string key = 1;
}

message ValueResponse {
    //Optional because if value is not found for key, then uses default empty string
    string value = 1;
}

message KeyValueRequest {
    string key = 1;
    string value = 2;
}

message SetResponse {
    //Optional because if unsuccessful, then it uses default value - false
    bool isSuccessful = 1;
}
